### PR TITLE
Add Model Registry Pagination

### LINKED ISSUE #93: [Model Registry] Add in pagination support
Add in server-side pagination for the model registry for use on the frontend.

### PR DESCRIPTION
**Pull request by StephenODea54**
Closes #93 

Adds in limit and offset arguments to support pagination.

### CODE REVIEW

File: apollo/src/graphql/model/modelQueries.ts [hunk 1]
```diff
@@ -6,13 +6,18 @@ builder.queryFields((t) => ({
         type: [MLModel],
         args: {
             modelName: t.arg.string(),
+            limit: t.arg.int({ required: true }),
+            offset: t.arg.int({ required: true }),
```
Comment (wetherc) [inline apollo/src/graphql/model/modelQueries.ts]: Could you add `defaultValue` to each of these and mark them as optional arguments? It might also make sense to convert `limit` into an enum for standard page sizes (maybe 10, 25, 50 or something like that)

Comment (StephenODea54) [reply apollo/src/graphql/model/modelQueries.ts]: Yeah I could add default values.

I was going to specify the limits on the frontend, but I can do them here as well.

Comment (wetherc) [reply apollo/src/graphql/model/modelQueries.ts]: Just given the variety of clients (frontend, python sdk, some madlad hitting the gql endpoint directly), it might make for a more consistent experience for things like that to be polite but firm suggestions maintained server-side

Comment (StephenODea54) [reply apollo/src/graphql/model/modelQueries.ts]: Fun fact, trying to define an enum of integer values in graphql is really hard lol

Comment (wetherc) [reply apollo/src/graphql/model/modelQueries.ts]: https://www.typescriptlang.org/docs/handbook/enums.html#numeric-enums

What's going wonky? Is Pothos just not playing nicely?

Comment (StephenODea54) [reply apollo/src/graphql/model/modelQueries.ts]: Nvm I'm the greatest engineer to ever live.

Comment (wetherc) [reply apollo/src/graphql/model/modelQueries.ts]: 😂 the duality of man

Comment (StephenODea54) [reply apollo/src/graphql/model/modelQueries.ts]: Okay, added a custom scalar type that takes in the values (10, 25, 50). If a request is made with a limit that is not null and is not within these values, an error is thrown.

Also created an array constant that holds the valid limits and hooked up a type to that so that they are always in sync.

image

image

File: apollo/src/types/Limit.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export const LIMITS = [10, 25, 50] as const;
+
+type LimitTuple = typeof LIMITS;
```
Comment (wetherc) [inline apollo/src/types/Limit.ts]: Sorry my brain is slow and dumb, could you ELI5 why not use an enum type instead? I'm just missing something obvious probably

Comment (StephenODea54) [reply apollo/src/types/Limit.ts]: Sorry just got back home.

I actually didn't see your comments before (fun fact, you have to refresh 🙃). I was having a hard time getting the enums to play nicely with Pothos, but I didn't know you could define numeric enums. I'm gonna give that a shot now and repush to this PR.

Comment (StephenODea54) [reply apollo/src/types/Limit.ts]: Okay, changed the `as const` to an actual enum. I kept the scalar and did not use a graphql enum because graphql enums cannot be referenced by their numeric values.

This means that the enum will resolve to the numeric value, but it cannot be referenced by its numeric value in the input. So if we had this enum:

```typescript
enum Limit {
    Ten = 10,
    TwentyFive = 25,
    Fifty = 50,
}
```

And we added this as an actual enum type in Pothos:

```typescript
builder.enumType(Limit, {
    name: 'Limit',
});
```

Then you would have to use, for example, `limit = 'Ten'` and not `limit = 10` in the query input.

Comment (StephenODea54) [reply apollo/src/types/Limit.ts]: Okay, changed the `as const` to an actual enum. I kept the scalar and did not use a graphql enum because graphql enums cannot be referenced by their numeric values.

This means that the enum will resolve to the numeric value, but it cannot be referenced by its numeric value in the input. So if we had this enum:

```typescript
enum Limit {
    Ten = 10,
    TwentyFive = 25,
    Fifty = 50,
}
```

And we added this as an actual enum type in Pothos:

```typescript
builder.enumType(Limit, {
    name: 'Limit',
});
```

Then you would have to use, for example, `limit = 'Ten'` and not `limit = 10` in the query input.

Comment (wetherc) [reply apollo/src/types/Limit.ts]: Ohhhhh weird and fun, cool cool

**Review by wetherc — APPROVED** [review]
